e2e/exec.go

```diff
@@ -0,0 +1,172 @@
+package e2e_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+const (
+	sshCommandTemplate                  = `echo '%s' > sshkey && chmod 0600 sshkey && ssh -i sshkey -o PasswordAuthentication=no -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o ConnectTimeout=5 azureuser@%s sudo`
+	listVMSSNetworkInterfaceURLTemplate = "https://management.azure.com/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/virtualMachineScaleSets/%s/virtualMachines/%d/networkInterfaces?api-version=2018-10-01"
+)
+
+type podExecResult struct {
+	exitCode       string
+	stderr, stdout *bytes.Buffer
+}
+
+func extractLogsFromVM(ctx context.Context, t *testing.T, vmssName string, sshPrivateKey string, opts *scenarioRunOpts) (map[string]string, error) {
+	privateIP, err := getVMPrivateIPAddress(ctx, opts.cloud, opts.suiteConfig.subscription, *opts.chosenCluster.Properties.NodeResourceGroup, vmssName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get private IP address of VM on VMSS %q: %s", vmssName, err)
+	}
+
+	commandList := map[string]string{
+		"/var/log/azure/cluster-provision.log": "cat /var/log/azure/cluster-provision.log",
+		"kubelet.log":                          "journalctl -u kubelet",
+	}
+
+	podName, err := getDebugPodName(opts.kube)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get debug pod name: %s", err)
+	}
+
+	var result = map[string]string{}
+	for file, sourceCmd := range commandList {
+		t.Logf("executing command on remote VM at %s of VMSS %s: %q", privateIP, vmssName, sourceCmd)
+
+		execResult, err := execOnVM(ctx, opts.kube, privateIP, podName, sshPrivateKey, sourceCmd)
+		if execResult != nil {
+			checkStdErr(execResult.stderr, t)
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		result[file] = execResult.stdout.String()
+	}
+	return result, nil
+}
+
+func extractClusterParameters(ctx context.Context, t *testing.T, kube *kubeclient) (map[string]string, error) {
+	commandList := map[string]string{
+		"/etc/kubernetes/azure.json":            "cat /etc/kubernetes/azure.json",
```

Comment:
did we extract just these 3 files from the cluster? my memory on ab e2e is fogged.